ktor-server/ktor-server-host-common/jvmAndNix/src/io/ktor/server/engine/CommandLine.kt

```diff
@@ -101,7 +101,7 @@ internal fun buildApplicationConfig(args: Array<String>): ApplicationConfig {
     val environmentConfig = getConfigFromEnvironment()
     val fileConfig = ConfigLoader.load(configPath)
 
-    return listOf(commandLineConfig, environmentConfig, fileConfig).merge()
+    return fileConfig.mergeWith(environmentConfig).mergeWith(commandLineConfig)
```

Comment:
it should be `withFallback` here, as the first configs should have priority.

Reply:
previous code was `listOf(commandLineConfig, environmentConfig, fileConfig).merge()`and `merge` was implemented through `foldRight`. For me it means that `commandLineConfig` is more priority than `environmentConfig` and the last is `fileConfig`. And that's why there is `mergeWith` where arg config has more priority from `fileConfig` to `commandLineConfig`